packages/mobile/src/geth/RNGethSigner.ts

```diff
@@ -0,0 +1,91 @@
+import { normalizeAddressWith0x } from '@celo/utils/src/address'
+import { Tx } from 'web3-core'
+import { Signer } from '@celo/contractkit/lib/wallets/signers/signer'
+import {
+  encodeTransaction,
+  extractSignature,
+  RLPEncodedTx,
+  rlpEncodedTx,
+} from '@celo/contractkit/lib/utils/signing-utils'
+import RNGeth from 'react-native-geth'
+import * as ethUtil from 'ethereumjs-util'
+
+const INCORRECT_PASSWORD_ERROR = 'could not decrypt key with given password'
+const currentTimeInSeconds = () => Math.round(Date.now() / 1000)
+
+/**
+ * Implements the signer interface on top of a RNGeth instance
+ */
+export class RNGethSigner implements Signer {
```

Comment:
could we merge `RNGethSigner` and `RNGethWallet` with `RpcWallet` and `RpcSigner` somehow?
it seems like the `RNGeth` interface is similar
there is also no other usage of `RpcWallet` and `RpcSigner` besides the wallet atm

Reply:
I don't see why they should be merged tbh, one is using a web3 instance to do RPC calls and one is calling a native module via a bridge. We could potentially get rid of `RpcWallet` and only have the `RNGethWallet` which btw I'd rename to `GethBridgeWallet` or something a bit more clear.